### PR TITLE
Update api.gateway.kyma-project.io CRD documentation with mTLS 

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Update documentation of CRD api.gateway.kyma-project.io with mTLS

**Related issue(s)**
See also #3541

### CODE REVIEW

File: docs/api-gateway/docs/06-01-api.md [hunk 1]
```diff
@@ -35,8 +35,10 @@ This table lists all the possible parameters of a given resource together with t
 | Field   |      Mandatory      |  Description |
 |----------|:-------------:|------|
 | **metadata.name** |    **YES**   | Specifies the name of the exposed API |
-| **service.name**, **service.port** | **YES** | Specifies the name and the communication port of the exposed service. |
+| **spec.service.name**, **spec.service.port** | **YES** | Specifies the name and the communication port of the exposed service. |
 | **spec.hostname** | **YES** | Specifies the service's external inbound communication address. |
+| **spec.authenticationEnabled** | **NO** | If `false`, disables generation of the Istio Authentication Policy for the service entirely. Defaults to `true`. |
```
Comment (kubadz) [inline docs/api-gateway/docs/06-01-api.md]: If `true` and `spec.authentication` is set, authentication settings from `spec.authentication` will be used. 
If `true` and `spec.authentication` is not set or set to `[]`, authentication will be enabled and default authentication settings specified in api-controller environment variables will be used. 
If `false` authentication will be disabled and `spec.authentication` will be ignored.
If not set, authentication will depend only on `spec.authentication` field.

Comment (Tomasz-Smelcerz-SAP) [reply docs/api-gateway/docs/06-01-api.md]: IMO this kind of description is too long here. I was very brief on purpose here. Perhaps we should make a dedicated section to document that behaviour.

Comment (kubadz) [reply docs/api-gateway/docs/06-01-api.md]: It is quite long, but your brief description is IMHO not sufficient, as that flag is not only used to disable auth but, above all, it is used to enable default auth. 
Your description is also wrong, as it doesn't default to `true`. `not set` is not equal to `true` in that case.

File: docs/api-gateway/docs/06-01-api.md [hunk 2]
```diff
@@ -35,12 +35,24 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.disableIstioAuthPolicyMTLS** | **NO** | Valid only if authentication is enabled. If `true`, disables the mTLS feature of the Istio Authentication Policy that secures the service. Defaults to `false`. |
 | **spec.authentication** | **NO** | Allows to specify an array of authentication policies that secure the service. |
 | **authentication.type** | **YES** | Specifies the desired authentication method that secures the exposed service. |
 | **authentication.jwt.issuer**, **authentication.jwt.jwksUri** | **YES** | Specifies the issuer of the tokens used to access the services, as well as the JWKS endpoint URI. |
 
+### Usage of authenticationEnabled parameter
+
+The `spec.authenticationEnabled` parameter controls if, and how, the API-Controller creates an Istio Authentication Policy that secures the exposed service.
+If authentication is enabled, `spec.authenticationEnabled` parameter sets conditions for the value of `origins` attribute of the corresponding Istio Authentication Policy.
```
Comment (tomekpapiernik) [inline docs/api-gateway/docs/06-01-api.md]: This sentence is redundant considering the bullets below and the information they give. Please remove it.

File: docs/api-gateway/docs/06-01-api.md [hunk 3]
```diff
@@ -35,12 +35,24 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.hostname** | **YES** | Specifies the service's external inbound communication address. |
+| **spec.authenticationEnabled** | **NO** | Controls the authentication attributes of the Istio Authentication Policy that secures the service. See the usage-of-authenticationEnabled-parameter section. |
+| **spec.disableIstioAuthPolicyMTLS** | **NO** | Valid only if authentication is enabled. If `true`, disables the mTLS feature of the Istio Authentication Policy that secures the service. Defaults to `false`. |
 | **spec.authentication** | **NO** | Allows to specify an array of authentication policies that secure the service. |
 | **authentication.type** | **YES** | Specifies the desired authentication method that secures the exposed service. |
 | **authentication.jwt.issuer**, **authentication.jwt.jwksUri** | **YES** | Specifies the issuer of the tokens used to access the services, as well as the JWKS endpoint URI. |
 
+### Usage of authenticationEnabled parameter
+
+The `spec.authenticationEnabled` parameter controls if, and how, the API-Controller creates an Istio Authentication Policy that secures the exposed service.
```
Comment (tomekpapiernik) [inline docs/api-gateway/docs/06-01-api.md]:
```suggestion
The `spec.authenticationEnabled` parameter controls if, and how, the API Controller creates an Istio Authentication Policy that secures the exposed service.
```

File: docs/api-gateway/docs/06-01-api.md [hunk 4]
```diff
@@ -35,12 +35,24 @@ This table lists all the possible parameters of a given resource together with t
...
+If authentication is enabled, `spec.authenticationEnabled` parameter sets conditions for the value of `origins` attribute of the corresponding Istio Authentication Policy.
+The behavior of the `authenticationEnabled` parameter is following:
```
Comment (tomekpapiernik) [inline docs/api-gateway/docs/06-01-api.md]:
```suggestion
These are the possible scenarios depending on the values the the **spec.authenticationEnabled** parameters takes: 
```

File: docs/api-gateway/docs/06-01-api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply docs/api-gateway/docs/06-01-api.md]: Done.

File: docs/api-gateway/docs/06-01-api.md [hunk 5]
```diff
@@ -35,12 +35,24 @@ This table lists all the possible parameters of a given resource together with t
...
+- if `false`, authentication is disabled and Authentication Policy is not created.
+- if not provided and `spec.authentication` is not provided or empty, authentication is disabled.
+- if not provided and `spec.authentication` is not empty, authentication is enabled. Entries from `spec.authentication` parameter are used to create the value of the `origins` attribute.
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply docs/api-gateway/docs/06-01-api.md]: Done.

File: docs/api-gateway/docs/06-01-api.md [hunk 6]
```diff
@@ -35,12 +35,24 @@ This table lists all the possible parameters of a given resource together with t
 | Field   |      Mandatory      |  Description |
 |----------|:-------------:|------|
 | **metadata.name** |    **YES**   | Specifies the name of the exposed API |
-| **service.name**, **service.port** | **YES** | Specifies the name and the communication port of the exposed service. |
+| **spec.service.name**, **spec.service.port** | **YES** | Specifies the name and the communication port of the exposed service. |
 | **spec.hostname** | **YES** | Specifies the service's external inbound communication address. |
+| **spec.authenticationEnabled** | **NO** | Controls the authentication attributes of the Istio Authentication Policy that secures the service. See [this](usage-of-authenticationenabled-parameter) section for more details. |
```
Comment (tomekpapiernik) [inline docs/api-gateway/docs/06-01-api.md]:
```suggestion
| **spec.authenticationEnabled** | **NO** | Controls the authentication attributes of the Istio Authentication Policy that secures the service. See [this](#usage-of-authenticationenabled-parameter) section for more details. |
```

**Pull request merged**